doc/Developing/Validating-Code.md

```diff
@@ -0,0 +1,42 @@
+source: Developing/Validating-Code.md
+
+As part of the pull request process with GitHub we run some automated build tests to ensure that 
+the code is error free, standards [compliant](http://docs.librenms.org/Developing/Code-Guidelines/)
+and our test suite builds successfully.
+
+Rather than submit a pull request and wait for the results, you can run these checks yourself to ensure 
+a more seamless merge.
+
+> All of these commands should be run from within the librenms directory and can be run as the librenms user 
+unless otherwise noted.
+
+### Syntax checks
+If you run PHP 5.3, 5.4 or HHVM please run:
+
+`find . -regextype posix-extended -regex "\./(lib/influxdb-php|vendor)" -prune -o -name "*.php" -print0 | xargs -0 -n1 -P8 php -l | grep -v '^No syntax errors detected' ; test $? -eq 1`
```

Comment:
I think we should move these into a file in scripts and then it would ease running it and keep updating things to a central location.